commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -334,6 +334,13 @@ public String toString() {
       return toStringExcludingParam(WEIGHT_PARAM_NAME);
    }
 
+   /**
+    * @return true if the storage type allows queries (indexed or non-indexed).
+    */
+   public boolean isQueryable() {
```

Comment:
This method does not really belong here. What is queryable and what isn't depends only on the query engine. If we're going to hardcode that then at lest let's do it in a not so public place :)This is only called by the BaseRemoteQueryManager so I'd move it there.

Reply:
Recently some code in the REST API needed to figure out if the format was queryable, this is something that often pops up. I'm OK with moving it somewhere else, what do you suggest?

Reply:
I will move this to the correct place, since config alone is not enough to detect if the cache is queryable, the storage is calculated at runtime